gammapy/catalog/hawc.py

```diff
@@ -127,14 +132,61 @@ def spectral_models(self):
         """
         models = [self._get_spectral_model(0)]
 
-        if self.n_spectra == 2:
+        if self.n_models == 2:
             models.append(self._get_spectral_model(1))
 
         return models
 
-    # TODO: add spatial models
+    def _get_spatial_model(self, idx):
+        d = self.data
+        label = f"spec{idx}_"
+
+        r_0 = d[label + "radius"]
+        if r_0 != 0.0:
+            model = DiskSpatialModel(d["glon"], d["glat"], r_0, frame="galactic")
+        else:
+            model = PointSpatialModel(d["glon"], d["glon"], frame="galactic")
```

Comment:
Typo: you filled "glon" for "lat". I noticed by adding a test case checking the parameter value in the model.